src/ptr_array.rs

```diff
@@ -0,0 +1,218 @@
+// Copyright 2019, The Gtk-rs Project Developers.
+// See the COPYRIGHT file at the top-level directory of this distribution.
+// Licensed under the MIT license, see the LICENSE file or <http://opensource.org/licenses/MIT>
+
+use glib_sys;
+use std::fmt;
+use translate::*;
+
+glib_wrapper! {
+    pub struct PtrArray(Shared<glib_sys::GPtrArray>);
```

Comment:
If we want to have proper `GPtrArray` bindings then this needs to generic over the contained pointer type and have the corresponding memory management handling for that type.

Unless you want to go that route, which is quite some effort, I'd recommend not having this part here but only the trait impls below for converting between pointer array and `Vec<T>` / `&[T]`. You can also move those trait impls into `translate.rs` next to the ones for `GList` and friends then.

Reply:
Now everything is moved to translate.rs